main.cpp

```diff
@@ -102,158 +106,138 @@ good to go!
+    DoubleType& subtract(double rhs );
+    DoubleType& multiply(double rhs );
+    DoubleType& divide(double rhs );
 };
 
 struct IntType
 {
-    int add( int lhs, int rhs );
-    int subtract( int lhs, int rhs );
-    int multiply( int lhs, int rhs );
-    int divide( int lhs, int rhs );
+    int* value = nullptr;
+    
+    IntType(int i_) : value(new int(i_)) {}
+    ~IntType()
+    {
+        delete value;
+    }
+
+    IntType& add(int rhs );
+    IntType& subtract(int rhs );
+    IntType& multiply(int rhs );
+    IntType& divide(int rhs );
 };
 
-float FloatType::add( float lhs, float rhs )
+
+FloatType& FloatType::add(float rhs)
 {
-    return lhs + rhs;
+    *value += rhs;
+    return *this;
 }
 
-float FloatType::subtract( float lhs, float rhs )
+FloatType& FloatType::subtract(float rhs)
 {
-    return lhs - rhs;
+    *value -= rhs;
+    return *this;
 }
 
-float FloatType::multiply( float lhs, float rhs )
+FloatType& FloatType::multiply(float rhs)
 {
-    return lhs * rhs;
+    *value *= rhs;
+    return *this;
 }
 
-float FloatType::divide( float lhs, float rhs )
+FloatType& FloatType::divide(float rhs)
 {
-    if ( rhs == 0.0f )
+    if (rhs == 0.0f)
         std::cout << std::endl << "warning, floating point division by zero returns 'inf' !" << std::endl;
-    return lhs / rhs;
+    *value /= rhs;
+    return *this;
 }
 
-double DoubleType::add( double lhs, double rhs )
+
+DoubleType& DoubleType::add(double rhs )
 {
-    return lhs + rhs;
+    *value += rhs;
+    return *this;
 }
 
-double DoubleType::subtract( double lhs, double rhs )
+DoubleType& DoubleType::subtract(double rhs )
 {
-    return lhs - rhs;
+    *value -= rhs;
+    return *this;
 }
 
-double DoubleType::multiply( double lhs, double rhs )
+DoubleType& DoubleType::multiply(double rhs )
 {
-    return lhs * rhs;
+    *value *= rhs;
+    return *this;
 }
 
-double DoubleType::divide( double lhs, double rhs )
+DoubleType& DoubleType::divide(double rhs )
 {
     if ( rhs == 0.0 )
         std::cout << std::endl << "warning, floating point division by zero returns 'inf' !" << std::endl;
-    return lhs / rhs;
+    *value = *value / rhs;
+    return *this;
 }
 
-int IntType::add( int lhs, int rhs )
+IntType& IntType::add(int rhs )
 {
-    return lhs + rhs;
+    *value += rhs;
+    return *this;
 }
 
-int IntType::subtract( int lhs, int rhs )
+IntType& IntType::subtract(int rhs )
 {
-    return lhs - rhs;
+    *value -= rhs;
+    return *this;
 }
 
-int IntType::multiply( int lhs, int rhs )
+IntType& IntType::multiply(int rhs )
 {
-    return lhs * rhs;
+    *value *= rhs;
+    return *this;
 }
 
-int IntType::divide( int lhs, int rhs )
+IntType& IntType::divide(int rhs )
 {
     if ( rhs == 0 )
     {
         std::cout << "error, integer division by zero will crash the program!" << std::endl << "returning lhs" << std::endl;
-        return lhs;
+        return *this;
     }
-    
-    return lhs / rhs;
+    *value /= rhs;
+    return *this;
```

Comment:
D.R.Y.  
you have 2 `return *this;` statements in this block.
use `else { }` to eliminate the extra `return *this;`
- [ ] revise